ocaml/typing/typecore.ml

```diff
@@ -6953,52 +6983,51 @@ and type_cases
             correct_levels ty_res
           else ty_res in
         let guard, exp =
-          match pc_guard with
-            | None | Some (Guard_predicate _) ->
+          match pc_rhs with
+            | Psimple_rhs rhs | Pboolean_guarded_rhs { pbg_rhs = rhs; _ } ->
```

Comment:
Let's fix the indentation of the `|` while we're here.

Reply:
Sure, done!